lib/cucumber/formatter/junit.rb

```diff
@@ -25,155 +21,134 @@ def initialize(runtime, io, options)
         @options = options
       end
 
-      def before_feature(feature)
-        @current_feature = feature
-        @failures = @errors = @tests = @skipped = 0
-        @builder = Builder::XmlMarkup.new( :indent => 2 )
-        @time = 0
+      def before_test_case(test_case)
+        if not same_feature_as_previous_test_case?(test_case.feature)
```

Comment:
[How to use Ruby's English and/or operators without going nuts](http://devblog.avdi.org/2014/08/26/how-to-use-rubys-english-andor-operators-without-going-nuts/) explains why the `if not` here really should be `if !`